src/main/webapp/app/project-select/project-select.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { Project, SortDirection, SortBy, ProjectComparators } from '../../model/project';
+import { request, fakeProjects, defaultColors } from '../../utils';
+import { faArrowDown, faArrowUp, faCircle } from '@fortawesome/free-solid-svg-icons';
+
+@Component({
+  selector: 'project-select',
+  templateUrl: './project-select.component.html',
+  styleUrls: ['./project-select.component.css']
+})
+/** Component which lets users select which projects to display on the graph */
+export class ProjectSelectComponent implements OnInit {
+  /** All projects we have access to */
+  public projects: Project[];
+  /** All projects that are currently selected */
+  public activeProjects: Set<Project>;
+  /** The sort direction */
+  public sortDirection: SortDirection;
+  /** The field sorting by */
+  public sortField: SortBy;
+
+  // Convenience selectors for the DOM
+  public iamBindings = SortBy.IAM_BINDINGS;
+  public projectName = SortBy.NAME;
+  public projectId = SortBy.PROJECT_ID;
+  public projectNumber = SortBy.PROJECT_NUMBER;
+
+  // #region Fontawesome icons
+  faIamArrow;
+  faNameArrow;
+  faProjectIdArrow;
+  faProjectNumberArrow;
+  faCircle;
+  // #endregion
+
+  @Output()
+  public changeProjects = new EventEmitter<Project[]>();
+
+  constructor() {
+    this.activeProjects = new Set();
+    this.faCircle = faCircle;
+
+    this.sortDirection = SortDirection.DESCENDING;
+    this.sortField = SortBy.IAM_BINDINGS;
+    this.setSortIcons();
+  }
+
+  /** Returns the color associated with the given project */
+  getColor(project: Project): string {
```

Comment:
In this case I don't think so, since this is being called directly by the Angular template and it's only function basically is to gray out the selector when necessary.